packages/quantic/force-app/main/default/lwc/quanticResultCopyToClipboard/quanticResultCopyToClipboard.js

```diff
@@ -0,0 +1,125 @@
+/** @typedef {import("coveo").InsightEngine} InsightEngine */
+
+/**
+ * The `QuanticResultCopyToClipboard` component allows the end user to copy a result to clipboard.
+ * @category Insight Panel
+ * @example
+ * <c-quantic-result-copy-to-clipboard engine-id={engineId} result={result} label="Copy" success-label="Copied!" text-template="${title}\n${clickUri}"></c-quantic-result-copy-to-clipboard>
+ */
+export default class QuanticResultCopyToClipboard extends LightningElement {
+  labels = {
+    copy,
+    copied,
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The result to copy.
+   * @api
+   * @type {Result}
+   */
+  @api result;
+  /**
+   * The label to be displayed in the tooltip of the button.
+   * @api
+   * @type {string}
+   */
+  @api label = this.labels.copy;
+  /**
+   * The label to be displayed in the tooltip of the button when the action is successful.
+   * @api
+   * @type {string}
+   */
+  @api successLabel = this.labels.copied;
+  /**
+   * The template used to generate the text to copy to clipboard.
+   * @api
+   * @type {string}
+   */
+  @api textTemplate = '${title}\n${clickUri}';
+
+  /** @type {object} */
+  actions;
+  /** @type {string} */
+  displayedLabel;
+  /** @type {InsightEngine} */
+  engine;
+
+  connectedCallback() {
+    this.displayedLabel = this.label;
+    registerComponentForInit(this, this.engineId);
+    this.addEventListener(
+      'quantic__copytoclipboard',
+      this.handleCopyToClipBoard
+    );
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener(
+      'quantic__copytoclipboard',
+      this.handleCopyToClipBoard
+    );
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  initialize = (engine) => {
+    this.engine = engine;
+    this.headless = getHeadlessBundle(this.engineId);
+
+    this.actions = {
+      ...this.headless.loadInsightAnalyticsActions(engine),
+    };
+  };
+
+  /**
+   * Performs the copy to clipboard action.
+   * @param {CustomEvent} event
+   */
+  handleCopyToClipBoard = (event) => {
+    event.stopPropagation();
+    const {setLoading, result} = event.detail;
+    setLoading(true);
+    const resultText = buildTemplateTextFromResult(this.textTemplate, result);
+
+    copyToClipboard(resultText)
+      .then(() => {
+        setLoading(false);
+        this.engine.dispatch(this.actions.logCopyToClipboard(this.result));
+        this.displayedLabel = this.successLabel;
+        this.resetOriginalLabel();
+      })
+      .catch((err) => {
+        setLoading(false);
+        console.error('Copy to clipboard action failed.', err);
+      });
+  };
+
+  /**
+   * Resets the original label after 500ms.
+   */
+  resetOriginalLabel() {
+    // eslint-disable-next-line @lwc/lwc/no-async-operation
+    setTimeout(() => {
+      this.displayedLabel = this.label;
+    }, 500);
```

Comment:
500 seems a bit short. maybe 1 second? I'm thinking of an old grandma user who likely wouldn't have time to notice a difference.